src/backends/caffe/caffelib.cc

```diff
@@ -181,6 +181,12 @@ namespace dd
     this->_mlmodel._model_template = model_tmpl;
     this->_logger->info("instantiating model template {}", model_tmpl);
 
+    if (!fileops::is_directory_writable(this->_mlmodel._repo))
```

Comment:
we probably could have this check higher up / more generic, maybe in `mlmodel` ?